engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java

```diff
@@ -0,0 +1,518 @@
+      allocate( argnr );
+
+      // Read them all...
+      for ( int a = 0; a < argnr; a++ ) {
+        variableName[a] = rep.getJobEntryAttributeString( id_jobentry, a, "variable_name" );
+        variableValue[a] = rep.getJobEntryAttributeString( id_jobentry, a, "variable_value" );
+        variableType[a] = getVariableType( rep.getJobEntryAttributeString( id_jobentry, a, "variable_type" ) );
+      }
+    } catch ( KettleException dbe ) {
+      throw new KettleException( BaseMessages.getString( PKG, "JobEntrySetVariables.Meta.UnableLoadRep", String
+        .valueOf( id_jobentry ), dbe.getMessage() ), dbe );
+    }
+  }
+
+  public void saveRep( Repository rep, IMetaStore metaStore, ObjectId id_job ) throws KettleException {
+    try {
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "replacevars", replaceVars );
+
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "filename", filename );
+      rep.saveJobEntryAttribute(
+        id_job, getObjectId(), "file_variable_type", getVariableTypeCode( fileVariableType ) );
+
+      // save the variableName...
+      if ( variableName != null ) {
+        for ( int i = 0; i < variableName.length; i++ ) {
+          rep.saveJobEntryAttribute( id_job, getObjectId(), i, "variable_name", variableName[i] );
+          rep.saveJobEntryAttribute( id_job, getObjectId(), i, "variable_value", variableValue[i] );
+          rep.saveJobEntryAttribute(
+            id_job, getObjectId(), i, "variable_type", getVariableTypeCode( variableType[i] ) );
+        }
+      }
+    } catch ( KettleDatabaseException dbe ) {
+      throw new KettleException( BaseMessages.getString( PKG, "JobEntrySetVariables.Meta.UnableSaveRep", String
+        .valueOf( id_job ), dbe.getMessage() ), dbe );
+    }
+  }
+
+  public Result execute( Result result, int nr ) throws KettleException {
+    result.setResult( true );
+    result.setNrErrors( 0 );
+    try {
+
+      List<String> variables = new ArrayList<String>();
+      List<String> variableValues = new ArrayList<String>();
+      List<Integer> variableTypes = new ArrayList<Integer>();
+
+      String realFilename = environmentSubstitute( filename );
+      if ( !Utils.isEmpty( realFilename ) ) {
+        try ( InputStream is = KettleVFS.getInputStream( realFilename );
+            // for UTF8 properties files
+            InputStreamReader isr = new InputStreamReader( is, "UTF-8" );
+            BufferedReader reader = new BufferedReader( isr );
+            ) {
+          Properties properties = new Properties();
+          properties.load( reader );
+          for ( Map.Entry entry : properties.entrySet() ) {
+            variables.add( (String) entry.getKey() );
+            variableValues.add( (String) entry.getValue() );
+            variableTypes.add( fileVariableType );
+          }
+        } catch ( Exception e ) {
+          throw new KettleException( BaseMessages.getString(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [KettleException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [365].  Consider adding a catch clause of this specific exception in the outer try statement.